inspirehep/modules/disambiguation/records.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Set of methods to create and update records."""
+
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+)
+
+from celery.utils.log import get_task_logger
+from flask import current_app, url_for
+
+from invenio_db import db
+
+from invenio_indexer.api import RecordIndexer
+
+from invenio_records import Record
+from invenio_records.signals import after_record_insert
+
+from inspirehep.dojson import utils as inspire_dojson_utils
+from inspirehep.modules.pidstore.minters import inspire_recid_minter
+
+from inspirehep.modules.disambiguation.receivers import (
+    append_new_record_to_queue,
+)
+
+logger = get_task_logger(__name__)
+
+
+def _get_author_schema():
+    """Generate schema URL to be inserted in $schema field."""
+    # FIXME: Use a dedicated method when #1355 will be resolved.
+    return url_for(
+        "invenio_jsonschemas.get_schema",
+        schema_path=current_app.config.get(
+            'DISAMBIGUATION_AUTHORS_SCHEMA_PATH'
+        )
+    )
+
+
+def create_author(profile):
+    """Create a new profile in authors index.
+
+    The method receives a dictionary representing an author.
+    Based on the values, it creates a dictionary in the invenio_records format.
+    After all the fields are processed, the method calls create_record
+    from invenio_records.api to put the new record.
+
+    :param profile:
+        A signature representing an author's to be created as a profile
+        with additional values for an experiment and a field_categories
+        taken from the hep paper.
```

Comment:
He can't, because https://github.com/inspirehep/inspire-next/pull/1353/files#diff-a9904eafdef51b485bb1dcb7e59343e8R129.

In fact, this method is assuming things that are not true in the schema, therefore reading and creating invalid author records.